src/appleseed.shaders/src/maya/as_maya_fractal.osl

```diff
@@ -0,0 +1,294 @@
+        float slidermin = 1.0,
+        float slidermax = 10.0,
+        string label = "Time Ratio",
+        string page = "Fractal Attributes"
+    ]],
+    float in_time = 0.0
+    [[
+        string maya_attribute_name = "time",
+        string maya_attribute_type = "float",
+        float slidermin = 0.0,
+        float slidermax = 100.0,
+        string label = "Time",
+        string page = "Fractal Attributes"
+    ]],
+
+    MAYA_COLORBALANCE_PARAMETERS,
+    MAYA_EFFECTS_PARAMETERS,
+    MAYA_UV_PARAMETERS,
+
+    output color out_outColor = color(0)
+    [[
+        string maya_attribute_name = "outColor",
+        string maya_attribute_type = "vector",
+        string label = "Output Color",
+        string widget = "null"
+    ]],
+    output float out_outAlpha = 1.0
+    [[
+        string maya_attribute_name = "outAlpha",
+        string maya_attribute_type = "float",
+        string label = "Output Alpha",
+        string widget = "null"
+    ]]
+)
+{
+    if (in_uvCoord[0] == OUTSIDE_UVFRAME || in_uvCoord[1] == OUTSIDE_UVFRAME)
+    {
+        out_outColor = in_defaultColor;
+        out_outAlpha = maya_luminance(in_defaultColor);
+    }
+    else
+    {
+        float st[2] = {in_uvCoord[0], in_uvCoord[1]};
+
+        float dsdt[2] = {
+            in_filter *
+                (in_uvFilterSize[0] == UNDEFINED_UVFILTER
+                    ? filterwidth(st[0])
+                    : in_uvFilterSize[0]),
+            in_filter *
+                (in_uvFilterSize[1] == UNDEFINED_UVFILTER
+                    ? filterwidth(st[1])
+                    : in_uvFilterSize[1])};
+
+        // The filter size factors seem to match Maya's result better.
+                
+        dsdt[0] += 2 * in_filterOffset;
+        dsdt[1] += 2 * in_filterOffset;
+
+#ifndef MAYA_FRACTAL_FREQ
+#define MAYA_FRACTAL_FREQ   14
+#endif
+
+        st[0] *= MAYA_FRACTAL_FREQ;
+        st[1] *= MAYA_FRACTAL_FREQ;
+
+        float filter_size = abs(max(dsdt[0], dsdt[1]));
+        
+        filter_size = clamp(
+            MAYA_FRACTAL_FREQ * filter_size,
+            MAYA_FRACTAL_FREQ * 1.0e-8,
+            0.5);
+
+        float bias_plus1, bias_minus1, bias_sqrplus1;
+
+        if (in_bias)
+        {
+            bias_plus1 = in_bias + 1;
+            bias_minus1 = in_bias - 1;
+            bias_sqrplus1 = sqr(bias_plus1);
+        }
+
+        float denom = (in_frequencyRatio == 2)
+            ? M_LN2
+            : log(max(1.01, in_frequencyRatio));
+
+        float iterate = clamp(
+            -log(filter_size) / denom, in_levelMin, in_levelMax);
+
+        int recursion_level = (int) floor(iterate);
+
+        float time_value = in_time, amplitude = in_amplitude, c = 0.0, val;
+        
+        for (int i = 0; i <= recursion_level; ++i)
+        {
+            if (i == recursion_level)
+            {
+                amplitude *= (iterate - (float) recursion_level);
+            }
+
+            if (!amplitude)
```

Comment:
We do allow omitting curly braces for single-line blocks, if you'd prefer that. But we're totally cool with curly braces in that case too. Whatever you prefer, just know that you have the choice :)